src/utils/read.js

```diff
@@ -7,16 +7,10 @@ const fs = require("fs");
  * var read = require('./helpers/fsExists')({});
  * read('.babelrc'); // file contents...
  */
-module.exports = function(cache) {
-  cache = cache || {};
```

Comment:
@danez main problems not in `restarting watch when the config changes`, main problems in watch graph (absence in dependencies configuration), it is break loader caches before `babel-loader`